Parser/tokenizer.c

```diff
@@ -1433,15 +1433,15 @@ tok_get(register struct tok_state *tok, char **p_start, char **p_end)
     /* Number */
     if (isdigit(c)) {
         if (c == '0') {
+            /* Hex, octal or binary -- maybe. */
+            char buf[100];
             if (Py_Py3kWarningFlag) {
-                if (PyErr_WarnExplicit(PyExc_Py3xWarning,
-                                    "octal literals are not supported in 3.x;\n" 
-                                    "drop the leading 0",
-                                    tok->filename, tok->lineno, NULL, NULL)) {
+                sprintf(buf, "use an 0o prefix for octal integers or drop the leading '%c'", (char *)c);
```

Comment:
`sprintf(3)` makes me nervous. I can see that the string fits in 100 characters, but someone else modifying this code later may accidentally overflow `buf`. Can we use `snprintf(3)` instead?

Also should the `100` be a defined constant?

Does `PyErr_WarnExplicit_WithFix()` guarantee not overflowing the buffer? I don't see how it could, as the length is not passed.

Reply:
We will avoid this for newer code, I see this convention used much in other parts of CPython. I just removed this line